src/input_common.h

```diff
@@ -111,6 +112,17 @@ class char_event_t {
     /// The type of event.
     char_event_type_t type;
 
+    /// Hackish: the input style, which describes how char events (only) are applied to the command
+    /// line. Note this is set only after applying bindings; it is not set from readb().
+    enum input_style_t : uint8_t {
```

Comment:
I was surprised for a bit because the enum is called `input_style_t` but it's used elsewhere using `char_event_t`.
It's obvious after a second look, but maybe it's better to make it a top-level `enum class` like the others above, since it needs to be public anyways.